packages/design-system/src/components/TextField/LabelMask.test.tsx

```diff
@@ -1,79 +1,127 @@
 import { useState } from 'react';
 import LabelMask from './LabelMask';
-import { DATE_MASK } from './index';
+import {
+  DATE_MASK,
+  PHONE_MASK,
+  ZIP_MASK,
+  SSN_MASK,
+  CURRENCY_MASK,
+  MaskFunction,
+} from './useLabelMask';
 import { render } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
 
-const TestLabelMask = () => {
+const allMasks = [
+  {
+    name: 'Date',
+    mask: DATE_MASK,
+    default: 'MM/DD/YYYY',
+    defaultData: '04072022',
+    defaultResult: '04/07/2022',
+    partialData: '041',
+    partialResult: '04/01/YYYY',
+  },
+  {
+    name: 'Social Security Number',
+    mask: SSN_MASK,
+    default: 'XXX-XX-XXXX',
+    defaultData: '123456789',
+    defaultResult: '123-45-6789',
+    partialData: '1234',
+    partialResult: '123-4X-XXXX',
+  },
+  {
+    name: 'Phone Number',
+    mask: PHONE_MASK,
+    default: 'XXX-XXX-XXXX',
+    defaultData: '1234567890',
+    defaultResult: '123-456-7890',
+    partialData: '1234',
+    partialResult: '123-4XX-XXXX',
+  },
+  {
+    name: 'US Zip Code',
+    mask: ZIP_MASK,
+    default: 'XXXXX',
+    defaultData: '12345',
+    defaultResult: '12345',
+    partialData: '123',
+    partialResult: '123XX',
+  },
+  {
+    name: 'US Currency',
+    mask: CURRENCY_MASK,
+    default: '$0.00',
+    defaultData: '12345.67',
+    defaultResult: '$12,345.67',
+    partialData: '123',
+    partialResult: '$123.00',
+  },
+];
+
+const TestLabelMask = (props: { mask: MaskFunction }) => {
+  const { mask } = props;
   const [value, setValue] = useState('');
   return (
-    <LabelMask labelMask={DATE_MASK}>
+    <LabelMask labelMask={mask}>
       <input type="text" value={value} onChange={(e) => setValue(e.currentTarget.value)} />
     </LabelMask>
   );
 };
 
-describe('Label mask', function () {
-  it('renders default date mask, MM/DD/YYYY, when no input value set', () => {
-    const { container, asFragment } = render(<TestLabelMask />);
-    const mask = container.querySelector('.ds-c-label-mask');
-    const input = container.querySelector('input');
-
-    expect(input.value).toBe('');
-    expect(mask.textContent).toContain('MM/DD/YYYY');
-
-    expect(asFragment()).toMatchSnapshot();
-  });
-  describe('updates label mask to reflect', () => {
-    it('complete input value set', () => {
-      const { container } = render(<TestLabelMask />);
-      const data = '12345678';
-      const maskText = '12/34/5678';
-
+allMasks.forEach((currentMask) => {
+  describe(`${currentMask.name} Label mask`, function () {
+    it(`renders default mask, ${currentMask.default}., when no input value set`, () => {
```

Comment:
There's a period here after the interpolated string I don't think should be there - it's showing up in snapshots too